all.sh

```diff
@@ -1,10 +1,42 @@
 #!/usr/bin/env bash
 
+usage() {
+  echo "Usage: $0 [--help] [-k FILTER]"
+  echo "  --help          Display this help message"
+  echo "  -k FILTER       Filter devices by name"
+  exit 1
+}
+
+while getopts ":hk:" opt; do
+  case ${opt} in
+    h)
+      usage
+      ;;
+    k)
+      MODULE_FILTER=$OPTARG
+      ;;
+    \?)
+      echo "Invalid option: $OPTARG" 1>&2
+      usage
+      ;;
+    :)
+      echo "Option -$OPTARG requires an argument." 1>&2
+      usage
+      ;;
+  esac
+done
+shift $((OPTIND -1))
+
```

Comment:
A nice touch would be to check if we have any leftover non-option argument and return error if that's the case:

```
while [[ $# -gt 0 ]]; do
  echo "Too many arguments: ${@}"
  usage
done
```